UI/window-basic-main.cpp

```diff
@@ -2139,6 +2139,9 @@ void OBSBasic::OBSInit()
 
 	/* ----------------------------- */
 	/* add custom browser docks      */
+#if BROWSER_AVAILABLE && YOUTUBE_ENABLED
```

Comment:
I _think_ this is not correct - at least `YOUTUBE_ENABLED` is only _defined_, but doesn't have an explicit value. So it is more correct to use `#if defined()`, because they don't carry any value (see line 2146 for the correct check).

Applies to the other occurrences of similar checks that just use `#if` for preprocessor definitions that don't carry a value.

Reply:
YOUTUBE_ENABLED may be defined as "0" (cmake `set(YOUTUBE_ENABLED OFF)`, for example). That's why all "YOUTUBE_ENABLED" under "#if" and not under "defined()". Undefined constants are replaced with "0" after all. So it's ok, for me.

![image](https://github.com/obsproject/obs-studio/assets/2684656/76269747-4f69-4dfb-9f0a-19c435122cb4)

Reply:
`set(YOUTUBE_ENABLED OFF)` does not propagate into C++ #if/#ifdef checks by itself. If `YOUTUBE_ENABLED` has been set to `OFF` in CMake, it is simply never propagated into C++ with `target_compile_definitions`.

Reply:
I suspect automoc generates `#define YOUTUBE_ENABLED 1` so that `#if YOUTUBE_ENABLED` works when the feature is enabled.
```text
$ grep -r YOUTUBE_ENABLED  build/
build/UI/obs_autogen/moc_predefs.h:#define YOUTUBE_ENABLED 1
```
When the feature is not enabled, GCC translate the undefined identifier to number `0`, according to [this document](https://gcc.gnu.org/onlinedocs/cpp/If.html). However, I cannot find how MSVC behaves when encountering the undefined identifier.
So, `#ifdef YOUTUBE_ENABLED` or `#if defined(YOUTUBE_ENABLED)` would be better.

Reply:
Guys, all similar OBS code uses "#if", not "#ifdef":
RESTREAM_ENABLED - "#if#"s, TWITCH_ENABLED - "#if", YOUTUBE_ENABLED - "#if"s (a lot of).

Do we really need to change only one of the 30 similar lines? The question is not about functionality, just a how code looks.

Reply:
> Guys, all similar OBS code uses "#if", not "#ifdef": RESTREAM_ENABLED - "#if#"s, TWITCH_ENABLED - "#if", YOUTUBE_ENABLED - "#if"s (a lot of).
> 
> Do we really need to change only one of the 30 similar lines? The question is not about functionality, just a how code looks.

@PatTheMav said in the first post:
> Applies to the other occurrences of similar checks that just use #if for preprocessor definitions that don't carry a value.

As far as I can tell, that means "this feedback applies to all other similar instances in the PR".

`#if` and `#ifdef` are not cosmetically different, they are functionally different. `#if defined` and `#ifdef` are functionally identical.

Reply:
`#if TWITCH_ENABLED` was implemented by a88b4402903. At that time, the macro was defined to `0` or `1` as below.
https://github.com/obsproject/obs-studio/blob/a88b440290372cc6699734c1df89027333436158/UI/ui-config.h.in#L19
Later, a commit 349372b3b39e2346952732ab14122d23c7ec7a57 removes the macro `TWITCH_ENABLED`, `RESTREAM_ENABLED`, and `YOUTUBE_ENABLED`. So, it's incorrect to use `#if TWITCH_ENABLED` now. I think we need to make a separate PR to fix the preprocess.

Reply:
@norihiro If there are existing instances outside of the changes in this PR that require fixups, then yes. @PatTheMav is pointing out that the places in this PR that add `#if YOUTUBE_ENABLED` are not correct with the way our current CMake passes the definition to code.

Reply:
So, should I change `#if BROWSER_AVAILABLE && YOUTUBE_ENABLED` to
1. `#if BROWSER_AVAILABLE && defined(YOUTUBE_ENABLED)`
2. `#if defined(BROWSER_AVAILABLE) && defined(YOUTUBE_ENABLED)`
4. nothing
5. nothing, but it will be new PR with changed all "#if YOUTUBE_ENABLED" to all "#ifdef YOUTUBE_ENABLED"
?

Reply:
> 2. `#if defined(BROWSER_AVAILABLE) && defined(YOUTUBE_ENABLED)`

This should be the appropriate one.

The merged PR #9242 has already fixed the existing preprocessor directives. You don't need to add a new wrong code.

Reply:
Option 2 is the only correct one.

Reply:
> Option 2 is the only correct one.

Changed.

Reply:
This is now resolved.

Reply:
There are still instances in this PR where you are adding `#if YOUTUBE_ENABLED` to C/C++ files. Please use `#if defined(YOUTUBE_ENABLED)` as previously instructed.

Reply:
Rechecked all "YOUTUBE_ENABLED" in source changes. Made `#ifdef`s.